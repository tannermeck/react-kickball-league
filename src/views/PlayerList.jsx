function PlayerList(){
    return (
        <>
            <h1>Hello from PlayerList</h1>
        </>
    )
}
export default PlayerList;