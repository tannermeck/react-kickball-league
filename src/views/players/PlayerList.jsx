import { useState, useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import { getPlayers } from '../../services/players';
import '../teams/teamdetail.css';

function PlayerList(){
    const [players, setPlayers] = useState([])
    const [loading, setLoading] = useState(true)

    useEffect(() => {
        async function getAllPlayers(){
            const playerList = await getPlayers();
            setPlayers(playerList)
            setLoading(false)
        }
        getAllPlayers()
    }, [])

    if (loading) return <h1>Loading Players...</h1>
    return (
        <>
            <h1 className='detailHeader'>Players:</h1>
            <ul>
                {players.map(player => {
                    return (
                    <NavLink key={player.id} to={`/players/${player.id}`}>
                        <li key={player.id}>{player.name}</li>
                    </NavLink>
                    )
                })}
            </ul>
        </>
    )
}
export default PlayerList;