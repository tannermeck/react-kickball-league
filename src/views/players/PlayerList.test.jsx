import { screen, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import PlayerList from './PlayerList';

const fakePlayer1 = {
    id: 1, 
    created_at: '2021-12-10T17:05:14.55043+00:00', 
    name: 'Amber', 
    position: 'organizer', 
    team_id: 17
}

const worker = setupServer(
    rest.get('https://bmugikqtddktbuecxyqi.supabase.co/rest/v1/players', (req, res, ctx) => {
        return res(
            ctx.json([fakePlayer1])
        )
    })
);

beforeAll(() => {
    worker.listen()
});
afterAll(() => {
    worker.close()
});

it('should render out a playerlist component', async () => { 
    const { container } = render(
                            <MemoryRouter>
                                <PlayerList />
                            </MemoryRouter>)
    const label = await screen.findByText('Players:', {exact: false})
    const player = await screen.findByText('Amber')
    expect(player).toBeInTheDocument();
    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});