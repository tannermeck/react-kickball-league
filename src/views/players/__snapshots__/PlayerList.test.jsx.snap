// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render out a playerlist component 1`] = `
<div>
  <h1
    class="detailHeader"
  >
    Players:
  </h1>
  <a
    class="newTeamLink"
    href="/players/new"
  >
    Add New Player
  </a>
  <ul>
    <div
      class="teamListContainer"
    >
      <a
        href="/players/1"
      >
        <li>
          Amber
        </li>
      </a>
      <button
        class="deleteButton"
      >
        Delete Amber
      </button>
      <a
        class="editTeam"
        href="/players/edit/1"
      >
        Edit Amber
      </a>
    </div>
  </ul>
</div>
`;
